crates/polars-stream/src/utils/byte_source.rs

```diff
@@ -0,0 +1,329 @@
+use std::ops::Range;
+use std::sync::Arc;
+
+use polars_error::{to_compute_err, PolarsResult};
+use polars_io::cloud::{
+    build_object_store, object_path_from_str, CloudLocation, CloudOptions, ObjectStorePath,
+    PolarsObjectStore,
+};
+use polars_utils::{_limit_path_len_io_err, no_call_const};
+
+use crate::async_executor;
+
+mod bytes_private {
+    use std::sync::Arc;
+
+    /// This is for keeping the buffers alive.
+    #[derive(Clone)]
+    #[allow(unused)]
+    pub(super) enum BytesKeepAlive {
+        Local(Arc<memmap::Mmap>),
+        Cloud(bytes::Bytes),
```

Comment:
This is basically the same as `polars_utils::mmap::MmapSlice`. I think you should use that.